### PR TITLE
OCPBUGS-41365: use appropriate images to avoid overrides in dataplane

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**: fixes image registry overrides being used in data plane.

Prereq: https://github.com/openshift/cluster-network-operator/pull/2507

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # [OCPBUGS-41365](https://issues.redhat.com/browse/OCPBUGS-41365)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/hostedclusterconfigoperator/cmd.go [hunk 1]
```diff
@@ -254,7 +254,7 @@ func (o *HostedClusterConfigOperator) Run(ctx context.Context) error {
 					Cache: map[string]*releaseinfo.ReleaseImage{},
 				},
 			},
-			RegistryOverrides: o.registryOverrides,
+			RegistryOverrides: nil,
```
Comment (sjenning) [inline control-plane-operator/hostedclusterconfigoperator/cmd.go:257]: why this change?

**Comment by Patryk-Stefanski** [comment]
kube-rbac-proxy container in the csi-driver-node pod also uses the registryOverrides need to talk with CSO team to add a new env var to control dataplane  and controlplane images individually
